span.go

```diff
@@ -49,22 +51,36 @@ type Span interface {
 	// may ignore the tag, but shall not panic.
 	SetTag(key string, value interface{}) Span
 
-	// LogEvent() is equivalent to
+	// LogFields and LogKV are two ways to record logging data about a Span.
+	// Both allow for timestamped key-value logging of arbitrary data. Neither
+	// intrinsically supports message formatting; in fact, formatted log
+	// messages are discouraged (though not disallowed) in OpenTracing.
 	//
-	//   Log(LogData{Event: event})
+	// LogFields is designed to be type-checked, efficient, yet a little
+	// cumbersome from the caller's perspective.
 	//
-	LogEvent(event string)
-
-	// LogEventWithPayload() is equivalent to
+	// LogKV is designed to minimize boilerplate and leads to concise, readable
+	// calling code; unfortunately this also makes it less efficient and less
+	// type-safe.
 	//
-	//   Log(LogData{Event: event, Payload: payload0})
+	// For example, the following are equivalent:
 	//
-	LogEventWithPayload(event string, payload interface{})
-
-	// Log() records `data` to this Span.
+	//    span.LogFields(
+	//        opentracing.LogString("request_path", request.Path()),
+	//        opentracing.LogInt("request_size", request.Size()))
 	//
-	// See LogData for semantic details.
-	Log(data LogData)
+	//    span.LogKV(
+	//        "request_path", request.Path(),
+	//        "request_size", request.Size())
+	//
+	// Also see Span.FinishWithOptions() and FinishOptions.BulkLogData.
+	LogFields(fields ...LogField)
```

Comment:
Well, it will *break* the previous signature, though at least it would do so obviously.

I am starting to feel as though the Go API has enough use that it's pretty shitty to break people's code without some grace period where the new and old are both supported.